automatic/kubernetes-cli/update.ps1

```diff
@@ -25,25 +25,46 @@ function global:au_SearchReplace {
 }
 
 function global:au_GetLatest {
+    # Only report the supported Kubernetes streams.
+    # NB Upstream Kubernetes only supports the last three minor versions.
+    #    However, they can also do pre-releases before a final version is
+    #    released, so we have to check for an extra changelogs/minor
+    #    version to be able to find all the supported streams.
+    # See https://github.com/kubernetes/sig-release/blob/master/release-engineering/versioning.md
+    $minor_versions_to_keep = 3
+
     $changelog_page = Invoke-WebRequest -Uri $changelogs -UseBasicParsing
-    #Only the latest three minor versions are supported: https://github.com/kubernetes/sig-release/blob/master/release-engineering/versioning.md
-    #However, there are prereleases before the version is released, so the latest four changelogs/minor versions need to be checked.
-    $minor_version_changelogs = $changelog_page.links | ? href -match "CHANGELOG-[\d\.]+\.md" | Select-Object -Expand href -First 4
-    
+    $minor_version_changelogs = $changelog_page.links `
+      | Where-Object href -match "CHANGELOG-(?<version>\d+\.\d+)\.md`$" `
+      | Select-Object -Expand href -First ($minor_versions_to_keep+1)
+
     $streams = @{}
-    
+
     $minor_version_changelogs | ForEach-Object {
-        $minor_version = ($_ -split "CHANGELOG-" | Select-Object -Last 1).trim(".md")
-        $minor_changelog_page = Invoke-WebRequest -UseBasicParsing -Uri ("https://github.com" + $_)
-        $url64 = $minor_changelog_page.links | ? href -match "/v(?<version>[\d\.]+)/kubernetes-client-windows-amd64.tar.gz" | Select-Object -First 1 -Expand href
+        if ($streams.Count -ge $minor_versions_to_keep) {
+          return
+        }
+        if ($_ -notmatch "CHANGELOG-(?<version>\d+\.\d+)\.md`$") {
+          return
+        }
+        $minor_version = $matches.version
+        $minor_changelog_page = Invoke-WebRequest -UseBasicParsing -Uri "https://github.com$_"
```

Comment:
Can we add a check before this line to see if the stream already contains the `$minor_version` so we don't need to do a web request when not needed.

Reply:
Just to note, this is more of a safe guard in case the changelog gets incorrectly updated at some point.

Reply:
Not sure you can prevent that with such a simple heuristic. For example, I guess this can go both ways. It can be flapping between wrong/right, so, I would not bother.

Reply:
Adding a simple if check if to see if the stream have already been added is mainly for the benefit of not aborting the entire collection.

If the changelog at some point will be incorrectly updated with the same major+minor version, then an exception will happen causing nothing to actually be opened, even the streams that was already added/will be added.

So, as I mentioned it is purely a safe guard so the updater won't throw an exception nothing else.

Basically the only thing that need to be added is:

```powershell
if ($streams.ContainsKey($minor_version)) {
  return
}
```

Reply:
Done. Please verify.

Reply:
The changes look good, thank you for adding the `if` check.